File: clion-plugin/src/main/java/com/linkyourbin/alfred/SwitchDevEnvProfileAction.java
package com.linkyourbin.alfred;

import com.intellij.openapi.actionSystem.AnAction;
import com.intellij.openapi.actionSystem.AnActionEvent;
import com.intellij.openapi.project.Project;
import com.intellij.openapi.ui.Messages;

import java.nio.file.Path;
import java.util.List;

public final class SwitchDevEnvProfileAction extends AnAction {
    @Override
    public void actionPerformed(AnActionEvent event) {
        Project project = event.getProject();
        if (project == null || project.getBasePath() == null) {
            Messages.showErrorDialog("Open a project first.", DevEnvProfiles.TITLE);
            return;
        }

        try {
            Path root = DevEnvProfiles.root(project);
            DevEnvProfiles.migrateProjectProfiles(root);
            Path profiles = DevEnvProfiles.profiles();
            List<Path> profileDirs = DevEnvProfiles.profileDirs(profiles);
            if (profileDirs.isEmpty()) {
                Messages.showInfoMessage(project, "No profiles found in " + profiles, DevEnvProfiles.TITLE);
                return;
            }

            String[] names = profileDirs.stream().map(path -> path.getFileName().toString()).toArray(String[]::new);
            String active = DevEnvProfiles.activeProfileName();
            String initial = active.isBlank() ? names[0] : active;
            String selected = Messages.showEditableChooseDialog(
                    "Choose a chip/dev env profile:",
                    DevEnvProfiles.TITLE,
                    Messages.getQuestionIcon(),
                    names,
                    initial,
                    null
            );
            if (selected == null || selected.isBlank()) {
                return;
            }

            DevEnvProfiles.applyProfileAfterExit(profiles.resolve(selected), root);
            Messages.showInfoMessage(
                    project,
                    "Profile staged: " + selected + "\nClose and reopen CLion to apply all saved project and IDE settings.",
                    DevEnvProfiles.TITLE
            );
        } catch (Exception e) {
            Messages.showErrorDialog(project, e.getMessage(), DevEnvProfiles.TITLE);
        }
    }
}

File: clion-plugin/src/main/kotlin/com/linkyourbin/alfred/SwitchDevEnvProfileAction.kt
package com.linkyourbin.alfred

import com.intellij.openapi.actionSystem.AnAction
import com.intellij.openapi.actionSystem.AnActionEvent
import com.intellij.openapi.ui.InputValidator
import com.intellij.openapi.ui.Messages

class SwitchDevEnvProfileAction : AnAction() {
    override fun actionPerformed(event: AnActionEvent) {
        val project = event.project
        if (project?.basePath == null) {
            Messages.showErrorDialog("Open a project first.", DevEnvProfiles.TITLE)
            return
        }

        try {
            val root = DevEnvProfiles.root(project)
            DevEnvProfiles.migrateProjectProfiles(root)
            val profiles = DevEnvProfiles.profiles()
            val profileDirs = DevEnvProfiles.profileDirs(profiles)
            if (profileDirs.isEmpty()) {
                Messages.showInfoMessage(project, "No profiles found in $profiles", DevEnvProfiles.TITLE)
                return
            }

            val names = profileDirs.map { it.fileName.toString() }.toTypedArray()
            val active = DevEnvProfiles.activeProfileName()
            val initial = active.ifBlank { names[0] }
            val selected = Messages.showEditableChooseDialog(
                "Choose a chip/dev env profile:",
                DevEnvProfiles.TITLE,
                Messages.getQuestionIcon(),
                names,
                initial,
                null as InputValidator?,
            )
            if (selected.isNullOrBlank()) {
                return
            }

            DevEnvProfiles.applyProfileAfterExit(profiles.resolve(selected), root)
            Messages.showInfoMessage(
                project,
                "Profile staged: $selected\nClose and reopen CLion to apply all saved project and IDE settings.",
                DevEnvProfiles.TITLE,
            )
        } catch (e: Exception) {
            Messages.showErrorDialog(project, e.message ?: e.toString(), DevEnvProfiles.TITLE)
        }
    }
}
